caffe2/contrib/tensorrt/tensorrt_tranformer.cc

```diff
@@ -0,0 +1,423 @@
+  }
+  std::vector<std::unique_ptr<NetDef>> nets;
+  nets.emplace_back(init_net);
+  nets.emplace_back(pred_net);
+  InferBlobShapesAndTypes(shape_hints_ordered, nets);
+  for (auto& net : nets) {
+    net.release();
+  }
+  std::unordered_map<std::string, TensorShape> shape_hints;
+  for (const auto& kv : shape_hints_ordered) {
+    shape_hints.emplace(kv.first, kv.second);
+  }
+
+  return shape_hints;
+}
+
+std::vector<std::string> FigureInputs(
+    const NetDef& pred_net,
+    int start,
+    int end,
+    const std::vector<OperatorDef>& new_ops,
+    const std::unordered_set<std::string>& weights,
+    std::unordered_set<std::string>* initialization_list) {
+  // TODO: cache this
+  std::unordered_set<std::string> boundary_inputs;
+  for (const auto& i : pred_net.external_input()) {
+    boundary_inputs.emplace(i);
+  }
+
+  //
+  for (const auto& op : new_ops) {
+    for (const auto& output: op.output()) {
+      boundary_inputs.emplace(output);
+    }
+  }
+
+  std::unordered_set<std::string> total_inputs;
+  std::vector<std::string> total_inputs_vec;
+  for(int i = start; i < end; ++i) {
+    const auto& op = pred_net.op(i);
+    for (const auto& input: op.input()) {
+      auto rt = total_inputs.emplace(input);
+      if (rt.second) {
+        if (weights.count(input)) {
+          // We add weights as inputs too
+          total_inputs_vec.emplace_back(input);
+          initialization_list->emplace(input);
+        } else if (boundary_inputs.count(input)) {
+          LOG(INFO) << "Adding boundary input: " << input;
+          total_inputs_vec.emplace_back(input);
+        }
+      }
+    }
+  }
+  return total_inputs_vec;
+}
+
+std::vector<std::string>
+FigureOutputs(const NetDef& pred_net, int start, int end) {
+  std::unordered_set<std::string> all_outputs;
+  std::vector<std::string> all_outputs_vec;
+  std::unordered_set<std::string> ext_outputs;
+  for (const auto& e: pred_net.external_output()) {
+    ext_outputs.emplace(e);
+  }
+  for (int i = start; i < end; ++i) {
+    const auto& op = pred_net.op(i);
+    for (const auto& output: op.output()) {
+      all_outputs.emplace(output);
+      all_outputs_vec.emplace_back(output);
+    }
+  }
+  std::unordered_set<std::string> referred_inputs;
+  for (int i = end; i < pred_net.op_size(); ++i) {
+    const auto& op = pred_net.op(i);
+    for (const auto& input: op.input()) {
+      referred_inputs.emplace(input);
+    }
+  }
+  // Remove the output that is
+  // 1. Not referred by the subsequent ops
+  // 2. Not in the external ouput of the net
+  all_outputs_vec.erase(
+      std::remove_if(
+          all_outputs_vec.begin(),
+          all_outputs_vec.end(),
+          [&ext_outputs, &referred_inputs](const std::string& output) {
+            return (
+                !referred_inputs.count(output) && !ext_outputs.count(output));
+          }),
+      all_outputs_vec.end());
+  return all_outputs_vec;
+}
+
+std::vector<::ONNX_NAMESPACE::ValueInfoProto> ConvertToValueInfo(
```

Comment:
I think it should be something different than `ONNX_NAMESPACE`, cc @bddppq

Reply:
? lf it comes from onnx proto, it should have `ONNX_NAMESPACE`. On the other hand, if this is messed up, it won't compile. :)

Reply:
IIUC, `ONNX_NAMESPACE` should be replaced with something else during build.